src/Sulu/Component/Rest/ListBuilder/Doctrine/DoctrineListBuilder.php

```diff
@@ -266,30 +261,72 @@ public function execute()
             return [];
         }
 
-        $this->queryBuilder = $this->createFullQueryBuilder(
+        $this->queryBuilder = $this->createNonGroupQueryBuilder(
             $this->em->createQueryBuilder()->from($this->entityName, $this->encodeAlias($this->entityName))
         );
 
         // now select all data
-        $this->assignJoins($this->queryBuilder);
+        $this->assignJoins($this->queryBuilder, $this->getNonGroupJoins());
 
         // use ids previously selected ids for query
         $select = $this->idField->getSelect();
         $this->queryBuilder->where($select . ' IN (:ids)')->setParameter('ids', $ids);
 
         $this->assignParameters($this->queryBuilder);
 
-        return $this->queryBuilder->getQuery()->getArrayResult();
+        $nonGroupResult = $this->queryBuilder->getQuery()->getArrayResult();
+
+        if (!$this->hasGroupingFieldDescriptor()) {
+            return $nonGroupResult;
+        }
+
+        $this->queryBuilder = $this->createGroupQueryBuilder(
+            $this->em->createQueryBuilder()->from($this->entityName, $this->encodeAlias($this->entityName))
+        );
+
+        // now select all data
+        $this->assignJoins($this->queryBuilder, $this->getGroupJoins());
+
+        // use ids previously selected ids for query
+        $select = $this->idField->getSelect();
+        $this->queryBuilder->where($select . ' IN (:ids)')->setParameter('ids', $ids);
+        $this->queryBuilder->indexBy($this->encodeAlias($this->entityName), $this->idField->getSelect());
+
+        $this->assignParameters($this->queryBuilder);
+
+        $groupResult = $this->queryBuilder->getQuery()->getArrayResult();
+
+        $result = [];
+        foreach ($nonGroupResult as $item) {
+            $result[] = \array_merge($item, $groupResult[$item[$this->idField->getName()]] ?? []);
```

Comment:
dumb question where is defined that `$groupResult` is indexed by `id`?

Reply:
https://github.com/sulu/sulu/pull/6998/files/de328f5e40aeffefbf39f61098c7f59f51838a9d#diff-d5e86b0228f4f8df3e86e01b2c60a132f15cbb4f24d5cda0f4c8f860c816577bR293